lib/src/commit_builder.rs

```diff
@@ -12,23 +12,29 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use uuid::Uuid;
+use std::sync::Arc;
 
 use crate::backend::{self, BackendResult, ChangeId, CommitId, ObjectId, Signature, TreeId};
 use crate::commit::Commit;
 use crate::repo::MutableRepo;
-use crate::settings::UserSettings;
+use crate::settings::{JJRng, UserSettings};
 
 #[must_use]
 pub struct CommitBuilder<'repo> {
     mut_repo: &'repo mut MutableRepo,
+    rng: Arc<JJRng>,
     commit: backend::Commit,
     rewrite_source: Option<Commit>,
 }
 
-// TODO: This function will be replaced in the next commit.
-pub fn new_change_id() -> ChangeId {
-    ChangeId::from_bytes(Uuid::new_v4().as_bytes())
+pub fn new_change_id(rng: &JJRng) -> ChangeId {
+    let random_bytes = rng.gen();
```

Comment:
I'm not sure which abstraction level is preferred:

If RNG is dedicated for change id (i.e. seeded with "debug.change-id-seed"), this function can be moved to `JJRng` wrapper, and the wrapper can be renamed accordingly.

If we want global RNG for various random thingy, this PR makes sense.

@martinvonz?

Reply:
My preference for this approach is also influenced by the opinion that there must be one shared RNG as far as possible, and care should be taken that it's never reset.

Reply:
I'm also not sure if we'll want any other randomness to use this randomness. I'm leaning towards making it specific to change ids for now since that's all we know we need.

We actually don't even need randomness for the change ids. It would be fine to let the "seed" be the top 64 bits of the change id and let the next 64 bits come from a per-run sequence number or something like that. All we need is for them each new change created in the tests to get a unique change id. We could still use randomness when the seed is not set. Anyway, generating change ids consistently by using a RNG is simple and makes the change ids in tests look more realistic.

Reply:
To be more specific: I think I would have personally put this function on `JJRng`, but I'm also fine with the current form. We can always change it later either way.